### PR TITLE
cdc: advancing resolved ts correctly for clusters with tiflash

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What is changed and how it works?

Issue Number: Close https://github.com/pingcap/tiflow/issues/4461

What's Changed:

```commit-message
cdc: advancing resolved ts correctly for clusters with tiflash
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/resolved_ts/src/advance.rs [hunk 1]
```diff
@@ -188,17 +188,18 @@ pub async fn region_resolved_ts_store(
         for peer in &peer_list {
             if peer.store_id == store_id && peer.id == leader_id {
                 resp_map.entry(region_id).or_default().push(store_id);
-                if peer_list.len() == 1 {
-                    valid_regions.insert(region_id);
-                }
-                continue;
+            } else if peer.get_role() != PeerRole::Learner {
```
Comment (overvenus) [inline components/resolved_ts/src/advance.rs:191]: Could you add some comments?

**Review by overvenus — APPROVED** [review]

File: components/resolved_ts/src/advance.rs [hunk 2]
```diff
@@ -238,52 +236,29 @@ pub async fn region_resolved_ts_store(
...
-                        tikv_clients.lock().await.remove(&to_store);
-                        return Err(Error::Other(box_err!(err)));
-                    }
-                };
-                Result::Ok((to_store, resp))
+                let resp = tokio::time::timeout(timeout, rpc)
+                    .map_err(|e| (to_store, true, format!("{}", e)))
+                    .await?
+                    .map_err(|e| (to_store, true, format!("{}", e)))?;
+                std::result::Result::Ok((to_store, resp))
```
Comment (hi-rustin) [inline components/resolved_ts/src/advance.rs:255]:
```suggestion
                Ok((to_store, resp))
```

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**